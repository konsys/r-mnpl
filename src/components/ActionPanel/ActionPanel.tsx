import React from "react";

export interface ActionPanelProps {
  text: string;
  onClick: () => void;
}
export const ActionPanel = (props: ActionPanelProps) => (
  <div className="TableAction-buttons">
    <div className="_action" onClick={props.onClick}>
      {props.text}
    </div>
  </div>
);
