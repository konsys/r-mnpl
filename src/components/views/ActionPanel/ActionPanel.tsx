import React from "react";

export interface ActionPanelProps {
  text: string;
  disabled: boolean;
  onClick: () => void;
}
export const ActionPanel = (props: ActionPanelProps) => (
  <div className="_action1" onClick={props.onClick}>
    {props.text}
  </div>
);
