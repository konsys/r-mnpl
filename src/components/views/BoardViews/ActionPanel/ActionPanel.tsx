import React from "react";

export interface ActionPanelProps {
  text: string;
  disabled: boolean;
  onClick: () => void;
}
export const ActionPanel = (props: ActionPanelProps) => (
  <div
    className={props.disabled ? "_action _locked" : "_action"}
    onClick={props.onClick}
  >
    {props.text}
  </div>
);
