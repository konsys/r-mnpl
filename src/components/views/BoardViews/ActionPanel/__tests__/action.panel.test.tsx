import { mount, shallow } from "enzyme";

import { ActionPanel } from "../ActionPanel";
import React from "react";
import renderer from "react-test-renderer";

describe("A suite", function () {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ActionPanel text={"test"} disabled={false} onClick={() => null} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render without throwing an error", () => {
    expect(
      mount(
        <ActionPanel text={"test"} disabled={false} onClick={() => null} />
      ).contains("test")
    ).toBeTruthy();
  });

  it("should render without throwing an error", () => {
    expect(
      shallow(
        <ActionPanel
          text={"test2"}
          disabled={false}
          onClick={() => console.log(2323)}
        />
      ).contains("test2")
    ).toBeTruthy();
  });

  it("should render without throwing an error", () => {
    expect(
      shallow(
        <ActionPanel text={"test"} disabled={true} onClick={() => null} />
      ).hasClass("_locked")
    ).toBeTruthy();
  });

  it("should render without throwing an error", () => {
    expect(
      shallow(
        <ActionPanel text={"test"} disabled={false} onClick={() => null} />
      ).hasClass("_locked")
    ).toBeFalsy();
  });

  it("should render an action panel", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <ActionPanel text={"test"} disabled={false} onClick={() => onChange(1)} />
    );
    expect(wrapper).toMatchSnapshot();

    wrapper.find("._action").simulate("click");
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it("should pass a selected value to the onChange handler", () => {
    const value = "2";
    const onChange = jest.fn();

    const wrapper = shallow(
      <ActionPanel text={"test"} disabled={false} onClick={onChange} />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper.find("div").simulate("click", value);

    expect(onChange).toBeCalledWith(value);
  });
});
