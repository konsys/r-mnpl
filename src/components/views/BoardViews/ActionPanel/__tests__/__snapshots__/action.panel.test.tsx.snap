// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite renders correctly 1`] = `
<div
  className="_action"
  onClick={[Function]}
>
  test
</div>
`;

exports[`A suite should pass a selected value to the onChange handler 1`] = `ShallowWrapper {}`;

exports[`A suite should render an action panel 1`] = `ShallowWrapper {}`;
